### PR TITLE
Add structure provider for string literals

### PR DESCRIPTION
**Pull request by alrz**
Fixes https://github.com/dotnet/roslyn/issues/24919

### CODE REVIEW

File: src/Features/CSharp/Portable/Structure/Providers/StringLiteralExpressoinStructureProvider.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+using System.Threading;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Options;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Structure;
+
+namespace Microsoft.CodeAnalysis.CSharp.Structure
+{
+    internal sealed class StringLiteralExpressoinStructureProvider : AbstractSyntaxNodeStructureProvider<LiteralExpressionSyntax>
```
Comment (genlu) [inline src/Features/CSharp/Portable/Structure/Providers/StringLiteralExpressoinStructureProvider.cs]: I think you need to add this new provider to the list in [`CSharpBlockStructureProvider.cs`](http://source.roslyn.io/#Microsoft.CodeAnalysis.CSharp.Features/Structure/CSharpBlockStructureProvider.cs)

File: src/EditorFeatures/CSharpTest/Structure/StringLiteralExpressionStructureTests.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+namespace Microsoft.CodeAnalysis.Editor.CSharp.UnitTests.Structure
+{
+    public class StringLiteralExpressionStructureTests : AbstractCSharpSyntaxNodeStructureTests<LiteralExpressionSyntax>
+    {
+        internal override AbstractSyntaxStructureProvider CreateProvider() => new StringLiteralExpressoinStructureProvider();
+
+        [Fact, Trait(Traits.Feature, Traits.Features.Outlining)]
+        public async Task TestVerbatimString()
+        {
+            await VerifyBlockSpansAsync(
```
Comment (genlu) [inline src/EditorFeatures/CSharpTest/Structure/StringLiteralExpressionStructureTests.cs]: Please also add tests for single line string literal and erroneous case like `string x = @";`

**Comment by genlu** [comment]
It'd be nice to have this in VB as well. Could you please add one? :)

File: src/EditorFeatures/CSharpTest/Structure/StringLiteralExpressionStructureTests.cs (hunk 2 above)
Comment (alrz) [reply src/EditorFeatures/CSharpTest/Structure/StringLiteralExpressionStructureTests.cs]: >single line string literal 

Looks like that is covered elsewhere for all structure providers, note we do report outlining for single line blocks, e.g. the following passes.
```cs
class C
{
    {|hint:$$namespace N{|textspan:{}|}|}
}
```

**Comment by genlu** [comment]
@jinujoseph @vatsalyaagrawal for approval

File: src/Features/CSharp/Portable/Structure/CSharpBlockStructureProvider.cs [hunk 3]
```diff
@@ -41,6 +41,7 @@ internal class CSharpBlockStructureProvider : AbstractBlockStructureProvider
             builder.Add<SimpleLambdaExpressionSyntax, SimpleLambdaExpressionStructureProvider>();
             builder.Add<StructDeclarationSyntax, TypeDeclarationStructureProvider, MetadataAsSource.MetadataTypeDeclarationStructureProvider>();
             builder.Add<SwitchStatementSyntax, SwitchStatementStructureProvider>();
+            builder.Add<LiteralExpressionSyntax, StringLiteralExpressoinStructureProvider>();
```
Comment (CyrusNajmabadi) [inline src/Features/CSharp/Portable/Structure/CSharpBlockStructureProvider.cs]: "Expressoin"

**Comment by genlu** [comment]
@CyrusNajmabadi More comments before I hit merge? :)

**Comment by CyrusNajmabadi** [comment]
1. I'd love a picture of what this looks like. 
2. Should we do template strings at the same time?  They can also be big and could benefit from this.

**Comment by CyrusNajmabadi** [comment]
@genlu Sorry.  "interpolated string".  i.e. `$"hello {world}"`.  JavaScript/TypeScript call it "template literal" and i transposed the names :)

**Comment by genlu** [comment]
@alrz I'm OK to merge this as is if you want to implement the interpolated string provider in a separate PR. Just let me know.

**Comment by alrz** [comment]
@genlu Yes, feel free to merge, I'll open a PR for interpolated strings.

**Comment by alrz** [comment]
@CyrusNajmabadi This is how it looks like

image

**Comment by CyrusNajmabadi** [comment]
👍 LGTM.  Thanks!

**Pull request merged**